pycytominer/tests/test_cyto_utils/test_cells.py

```diff
@@ -235,10 +235,12 @@ def test_SingleCells_count():
 
 def test_load_compartment():
     loaded_compartment_df = ap.load_compartment(compartment="cells")
-    pd.testing.assert_frame_equal(loaded_compartment_df, cells_df)
+    pd.testing.assert_frame_equal(
+        loaded_compartment_df, cells_df, check_dtype=False)
 
     # Test non-canonical compartment loading
-    pd.testing.assert_frame_equal(new_compartment_df, ap_new.load_compartment("new"))
+    pd.testing.assert_frame_equal(
+        new_compartment_df, ap_new.load_compartment("new"), check_dtype=False)
```

Comment:
I am surprised that this passes the tests, given this function:

```python
def is_feature_col(self, col):
    """Check if column is a feature."""
    return (
        col.startswith("Cell")
        or col.startswith("Cytoplasm")
        or col.startswith("Nuclei")
    )
```

Is it possible that all features prefixed with `new` are considered metadata features in your implementation?

Reply:
Fixed it by checking whatever is provided in `self.compartments`.